esp_idf/spotflow/src/configs/spotflow_config.c

```diff
@@ -0,0 +1,119 @@
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "logging/spotflow_log_backend.h"
+#include "logging/spotflow_log_cbor.h"
+#include "configs/spotflow_config.h"
+#include "configs/spotflow_config_cbor.h"
+#include "configs/spotflow_config_persistance.h"
+#include "configs/spotflow_config_net.h"
+#include "configs/spotflow_config_options.h"
+
+static void add_log_severity_to_reported_msg(struct spotflow_config_reported_msg* reported_msg);
+
+/**
+ * @brief Initilize the cloud configurable log level.
+ *
+ */
+void spotflow_config_init()
+{
+	struct spotflow_config_persisted_settings persisted_settings;
+
+	spotflow_config_persistence_try_init();
+	spotflow_config_persistence_try_load(&persisted_settings);
+
+	if ((persisted_settings.flags & SPOTFLOW_REPORTED_FLAG_MINIMAL_LOG_SEVERITY) != 0) {
+		spotflow_config_init_sent_log_level(persisted_settings.sent_log_level);
+	} else {
+		spotflow_config_init_sent_log_level_default();
+	}
+	spotflow_config_init_session();
+}
+
+/**
+ * @brief Initialize the sesson and send a intiail message with current configurations.
+ *
+ * @return int
+ */
+int spotflow_config_init_session()
+{
+	struct spotflow_config_reported_msg reported_msg = {
+		.flags = 0,
+	};
+	add_log_severity_to_reported_msg(&reported_msg);
+
+	int rc = spotflow_config_prepare_pending_message(&reported_msg);
+	if (rc < 0) {
+		SPOTFLOW_LOG(
+		    "Failed to prepare initial reported configuration response message: %d", rc);
```

Comment:
Please apply this minor fix - it's not a response. My bad, it's in Zephyr as well - can you fix it there as well?
```suggestion
		    "Failed to prepare initial reported configuration message: %d", rc);
```